spec/workers/commit_monitor_handlers/batch/github_pr_commenter/diff_content_checker_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+describe CommitMonitorHandlers::Batch::GithubPrCommenter::DiffContentChecker do
+  let(:batch_entry)        { BatchEntry.create!(:job => BatchJob.create!) }
+  let(:branch)             { create(:pr_branch) }
+  let(:diff)               { GitService::Diff.new(rugged_diff) }
+  let(:git_service_double) { double("GitService", :diff => diff) }
```

Comment:
This is what I mean: https://github.com/ManageIQ/miq_bot/blob/master/spec/workers/commit_monitor_handlers/batch/github_pr_commenter/diff_filename_checker_spec.rb#L4

Trying to avoid creating a ton of lets for things that are essentially "private" values that aren't used by specs directly.  I might even consider collapsing that entire rugged_diff definition into:

```ruby
let(:diff) do
  rugged_line_1  = double("RuggedLine", :addition? => true, :content => content_1, :new_lineno => 1) }
  rugged_line_2  = double("RuggedLine", :addition? => true, :content => content_2, :new_lineno => 2) }
  rugged_hunk_1  = double("RuggedHunk", :lines => [rugged_line_1, rugged_line_2]) }
  rugged_delta   = double("RuggedDelta", :new_file => {:path => file_path}) }
  rugged_patch_1 = double("RuggedPatch", :hunks => [rugged_hunk_1], :delta => rugged_delta) }
  rugged_diff    = double("RuggedDiff", :patches => [rugged_patch_1]) }
  GitService::Diff.new(rugged_diff)
end